ddl/rollingback.go

```diff
@@ -157,11 +157,35 @@ func rollingbackAddColumns(t *meta.Meta, job *model.Job) (ver int64, err error)
 }
 
 func rollingbackDropColumn(t *meta.Meta, job *model.Job) (ver int64, err error) {
-	tblInfo, colInfo, err := checkDropColumn(t, job)
+	tblInfo, colInfo, idxInfos, err := checkDropColumn(t, job)
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
 
+	if len(idxInfos) > 0 {
+		for _, indexInfo := range idxInfos {
+			originalState := indexInfo.State
+			switch indexInfo.State {
+			case model.StateWriteOnly, model.StateDeleteOnly, model.StateDeleteReorganization, model.StateNone:
```

Comment:
Why it's `StateNone`, it's possible to rollback.

Reply:
When there has index that should be dropped and the indexInfo.State is StateNone it means cannot rollback. This code is reference from `rollingbackDropIndex`. But when there has no index to be dropped, StateNone can be rollback.

Reply:
When `indexInfo.State` is `StateNone`, means there has delete range job submitted. So we cannot just rollback the `TableInfo` to old version, we may recreate the indices that dropped.

Reply:
Got it.